tortoise/sim/partition.go

```diff
@@ -0,0 +1,186 @@
+package sim
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/log"
+)
+
+// Frac is a shortcut for creating Fraction object.
+func Frac(nominator, denominator int) Fraction {
+	return Fraction{Nominator: nominator, Denominator: denominator}
+}
+
+// Fraction of something.
+type Fraction struct {
+	Nominator, Denominator int
+}
+
+func (f Fraction) String() string {
+	return fmt.Sprintf("%d/%d", f.Nominator, f.Denominator)
+}
+
+// SplitOpt is for configuring partition.
+type SplitOpt func(*splitConf)
+
+// WithPartitions configures number of miners in each partition
+// relative to original Generator.
+func WithPartitions(parts ...Fraction) SplitOpt {
+	return func(c *splitConf) {
+		c.Partitions = parts
+	}
+}
+
+func splitConfDefaults() splitConf {
+	return splitConf{
+		Partitions: []Fraction{Frac(1, 2), Frac(1, 2)},
+	}
+}
+
+type splitConf struct {
+	Partitions []Fraction
+	// TODO(dshulyak) support both transient and full partitions
+	//
+	// transient partitions can be configured without splitting state, in transient partitions
```

Comment:
is a transient partition like a malicious attack in the network?

Reply:
i was thinking more about just a regular network partition, but within single epoch. so there will be no new atx's and beacon remains the same